wikimedia/wikipedia-ios
# On This Day Widget

**Fixes Phabricator ticket:** 

### Notes
* 

### Test Steps
1. Test English. Test a non-English LTR language (like French). Test a RTL language (like Arabic). 
1. Ensure when you tap on the widget, it goes to appropriate page. And if you tap the article box on the large widget, it takes you there.
1. Try a language that doesn't support On this day. (Most of them do not support it.) It should give a nice error message.
1. Make text size smaller and larger, makes sure it's okay.
1. No connection: Since it will normally use the most recent valid hit, to test the "no internet" error message you should uninstall the app, turn on airplane mode, install the app, and add a widget. (And if this is the only possible way to see this error message, it's overkill. It's in there in case it ever tries to refresh when there is no internet, and it doesn't fall back on the last valid data.)
1. Placeholder text: Easiest way to test this putting `completion(OnThisDayData().placeholderEntry)` in for `completion` of the following block, then looking at it in the "add widget" system menu (from springboard):
```
func getSnapshot(in context: Context, completion: @escaping (OnThisDayEntry) -> Void) {
        dataStore.fetchLatestAvailableOnThisDayEntry(usingCache: context.isPreview) { entry in
            completion(entry)
        }
    }
```

### Screenshots/Videos *(if applicable)*
